workflow/controller/steps.go

```diff
@@ -112,9 +113,14 @@ func (woc *wfOperationCtx) executeSteps(ctx context.Context, nodeName string, tm
 				// This happens when there was `withItem/withParam` expansion.
 				// We add the aggregate outputs of our children to the scope as a JSON list
 				var childNodes []wfv1.NodeStatus
+				retryNodeNamePattern, _ := regexp.Compile(`.*\([0-9]+\)$`)
```

Comment:
Once the previous comment is addressed, this can likely use the exam same variable as it is in the same package, no?